ramlfications/parser/main.py

```diff
@@ -372,12 +372,70 @@ def create_resources(node, resources, root, parent):
             else:
                 child = _create_resource_node(name=k, raw_data=v, method=None,
                                               parent=parent, root=root)
-                resources.append(child)
+                if "type" in v:
+                    # There may be some more methods defined on the resource
+                    # type this one inherits from
+                    supertype = v["type"]
+                    resources = _create_supertype_resources(
+                        root, k, parent, supertype, resources)
+                else:
+                    resources.append(child)
             resources = create_resources(child.raw, resources, root, child)
 
     return resources
 
 
+def _create_supertype_resources(root, type_name, parent, supertype, resources):
+    """
+    Recursively traverses the inheritance tree for a resource via DFS to
+    find the resource endpoints associated with it.
+
+    :param RootNode root: The ``.raml.RootNode`` of the API
+    :param str type_name: The name of the resource type currently being
+                          inspected.
+    :param ResourceNode parent: parent node that inherited resource endpoints
+                                are to be attached to.
+    :param str supertype: The name of the supertype for the current node.
+    :param list resources: List of collected ``.raml.ResourceNode`` s
+    :returns: List of :py:class:`.raml.ResourceNode` objects.
+    """
+    resource_types = _get_resource_types_by_name(root, supertype)
+    for resource_type in resource_types:
+        resource_supertype = getattr(resource_type, "type", None)
+        if resource_supertype is not None:
+            resources = _create_supertype_resources(
+                root, type_name, parent, resource_supertype, resources)
+        resource = _create_resource_node(
+            name=type_name,
+            raw_data=resource_type.raw,
+            method=resource_type.method,
+            parent=parent,
+            root=root
+        )
+        resource.type = supertype
+        resources.append(resource)
+    return resources
+
+
+def _get_resource_types_by_name(root, name):
```

Comment:
This is more of a utility function, can we move it to `ramlfications/utils/parser.py` somewhere?